dig.go

```diff
@@ -285,10 +290,12 @@ func (c *Container) createInObject(t reflect.Type) (reflect.Value, error) {
 		e := edge{key: key{t: f.Type, name: f.Tag.Get(_nameTag)}, optional: isOptional}
 		v, err := c.get(e)
 		if err != nil {
-			return dest, fmt.Errorf(
-				"could not get field %v (edge %v) of %v: %v", f.Name, e, t, err)
+			if _, ok := err.(errMissing); ok {
+				v = reflect.Zero(f.Type)
```

Comment:
Wouldn't you rather do this in `get()` where you're returning `errMissing`?

```go
if err := c.contains(n.deps); err != nil {
  if e.optional { .. }
  else { .. }
}
```

That would actually let us drop the error type altogether.